default/AI/ProductionAI.py

```diff
@@ -66,22 +101,37 @@ def update_best_mil_ship_rating():
     cost = best_design.productionCost(fo.empireID(), build_choices[0])
     bestMilRatingsHistory[fo.currentTurn()] = (stats['overall'], cost)
 
+
 def cur_best_mil_ship_rating():
-    if fo.currentTurn() not in bestMilRatingsHistory:
-        update_best_mil_ship_rating()
-    return bestMilRatingsHistory[ fo.currentTurn()][0]
+    priority = EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_MILITARY
+    if priority in design_cache:  # use new framework
+        try:
+            return design_cache[priority][0][0]
+        except Exception:
+            print traceback.format_exc()
```

Comment:
you can use `print_exc` it print error to `stderr`.  It will be marked as error in logs.